db/transaction.go

```diff
@@ -2,18 +2,28 @@ package db
 
 import (
 	"errors"
+	"fmt"
 	"sync"
 	"sync/atomic"
 
 	"github.com/albrow/stringset"
 )
 
 var (
-	ErrDiscarded             = errors.New("transaction has already been discarded")
-	ErrCommitted             = errors.New("transaction has already been committed")
-	ErrConflictingOperations = errors.New("cannot perform more than one operation (insert/delete/update) on the same model within a transaction")
+	ErrDiscarded = errors.New("transaction has already been discarded")
+	ErrCommitted = errors.New("transaction has already been committed")
 )
 
+// ConflictingOperationsError is returned when two conflicting operations are attempted within the same
+// transaction
+type ConflictingOperationsError struct {
+	operation string
+}
+
+func (e ConflictingOperationsError) Error() string {
+	return fmt.Sprintf("cannot perform more than one operation (%s) on the same model within a transaction", e.operation)
```

Comment:
```suggestion
	return fmt.Sprintf("error on %s: cannot perform more than one operation on the same model within a transaction", e.operation)
```

Just a slight re-wording here. The current version doesn't make it clear whether `operation` is the first operation or the second one.